lib/her/model/associations/has_many_association.rb

```diff
@@ -85,7 +85,15 @@ def create(attributes = {})
         def fetch
           super.tap do |o|
             inverse_of = @opts[:inverse_of] || @parent.singularized_resource_name
-            o.each { |entry| entry.send("#{inverse_of}=", @parent) }
+
+            o.each do |entry|
+              entry.send("#{inverse_of}=", @parent)
+
+              # This fix requires ActiveModel 4.2+.
+              if entry.respond_to?(:clear_attribute_changes, true)
```

Comment:
nice 👍 

Once we update `ActiveModel` we should probably look at `clear_changes_information` after a record is fetched or reloaded and `changes_applied` after persistence.